web/modules/custom/collection/collection.behat.inc

```diff
@@ -157,6 +167,36 @@ class CollectionSubContext extends DrupalSubContextBase implements DrupalSubCont
   }
 
   /**
+   * Indexes items of entity type: collection.
+   *
+   * In addiction, views cache data are being cleared.
+   *
+   * @throws \Exception
+   *   When no items are indexed.
+   *
+   * @Given The collections are indexed
+   */
+  public function theCollectionsAreIndexed() {
+    /** @var String $id */
+    $id = 'collections';
+
+    /** @var String $views_id */
+    $views_id = 'collection';
+
+    /** @var \Drupal\search_api\IndexInterface $index */
+    $index = \Drupal\search_api\Entity\Index::load($id);
+    if (empty($index)) {
+      throw new \Exception("Collection index not found.");
+    }
+    $index->reindex();
+    $index->indexItems();
+
+    // Clear the views cache data.
+    $view_tags = \Drupal\views\Views::getView($views_id)->getCacheTags();
+    \Drupal\Core\Cache\Cache::invalidateTags($view_tags);
```

Comment:
I am not sure on this, it is just the way it worked for me. By default, I found that views save a result cache in "cache_data" table using the "config:views:view:<view_machine_name>" as tags. Even by disabling the cache in views settings, did not work for me when indexing items.
All in all, the test was successful at the first run, but the results would be cached so a manual cache clear or re-saving the view was needed. 
If you have any suggestion on this I am open but my tests would only work with this piece of code.